api/commands/prune.ts

```diff
@@ -75,6 +75,17 @@ function computeIncludedActionNames(
         }
       });
     }
+
+    // Add assertions which are direct dependents of actions.
+    const assertions = compiledGraph.assertions;
+    assertions.forEach(assertion => {
+      const matchingDependencyNames = !!assertion.dependencies?.length
+        ? utils.matchPatterns(assertion.dependencies, allActionNames)
+        : [];
+      if (!!matchingDependencyNames?.length) {
```

Comment:
this seems incorrect: if the assertion depends on 2 actions, 1 of which isn't included in the run, IMO the assertion shouldn't run.